pkg/controller/dns/recordset/managed.go

```diff
@@ -0,0 +1,181 @@
+	errGetDNSRecordSet    = "cannot get DNS RecordSet"
+	errDeleteDNSRecordSet = "cannot delete DNS RecordSet"
+)
+
+// Setup adds a controller that reconciles DNS RecordSets.
+func Setup(mgr ctrl.Manager, l logging.Logger, rl workqueue.RateLimiter, poll time.Duration) error {
+	name := managed.ControllerName(v1alpha1.RecordSetGroupKind)
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		WithOptions(controller.Options{
+			RateLimiter: ratelimiter.NewDefaultManagedRateLimiter(rl),
+		}).
+		For(&v1alpha1.RecordSet{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(v1alpha1.RecordSetGroupVersionKind),
+			managed.WithConnectionPublishers(),
+			managed.WithExternalConnecter(&connecter{client: mgr.GetClient()}),
+			managed.WithReferenceResolver(managed.NewAPISimpleReferenceResolver(mgr.GetClient())),
+			managed.WithPollInterval(poll),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type connecter struct {
+	client client.Client
+}
+
+func (c *connecter) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	creds, auth, err := azureclients.GetAuthInfo(ctx, c.client, mg)
+	if err != nil {
+		return nil, err
+	}
+	cl := dns.NewRecordSetsClient(creds[azureclients.CredentialsKeySubscriptionID])
+	cl.Authorizer = auth
+	return &external{
+		client: dnsclients.NewRecordSetClient(cl),
+	}, nil
+}
+
+type external struct {
+	client dnsclients.RecordSetAPI
+}
+
+func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
+	r, ok := mg.(*v1alpha1.RecordSet)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errNotDNSRecordSet)
+	}
+
+	az, err := e.client.Get(ctx, r)
+	if azureclients.IsNotFound(err) {
+		return managed.ExternalObservation{ResourceExists: false}, nil
+	}
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errGetDNSRecordSet)
+	}
+
+	r.Spec.ForProvider.Metadata = azureclients.LateInitializeStringMap(r.Spec.ForProvider.Metadata, az.Metadata)
+
+	dnsclients.UpdateRecordSetStatusFromAzure(r, az)
+
+	if r.Status.AtProvider.ProvisioningState != dnsclients.RecordSetSuccessfulState {
+		return managed.ExternalObservation{ResourceExists: true, ResourceUpToDate: true, ConnectionDetails: managed.ConnectionDetails{}}, nil
+	}
+
+	r.SetConditions(xpv1.Available())
+
+	o := managed.ExternalObservation{
+		ResourceExists:    true,
+		ResourceUpToDate:  dnsclients.RecordSetIsUpToDate(&r.Spec.ForProvider, az.RecordSetProperties),
+		ConnectionDetails: managed.ConnectionDetails{},
+	}
+
+	return o, nil
+}
+
+func (e *external) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) {
+	r, ok := mg.(*v1alpha1.RecordSet)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errNotDNSRecordSet)
+	}
+
+	r.Status.SetConditions(xpv1.Creating())
```

Comment:
nit: Managed reconciler should be doing this. Could you please check?

Reply:
I checked this. As you said that, managed reconciler does this operation. So I removed this line.